lib/serviceClient.ts

```diff
@@ -410,7 +410,7 @@ function createDefaultRequestPolicyFactories(credentials: ServiceClientCredentia
 
   factories.push(deserializationPolicy(options.deserializationContentTypes));
 
-  if (options.proxySettings) {
+  if (options.proxySettings || (options.proxySettings = getDefaultProxySettings())) {
```

Comment:
Assignment within an if-expression? Please pull that assignment out of the expression. ;)